frontend/components/utils/BidInput.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+    <div class="w-full inline-block relative flex-shrink-0">
+        <Tooltip :visible="!!error" placement="topLeft" :title="error">
+            <label>
+                <div v-if="!value" class="absolute text-right right-8 top-1">
+                    <format-currency v-if="totalPrice && !disabled" :value="totalPrice" />
+                    <span v-else class="opacity-50">UNKNOWN</span>
+                </div>
+                <span class="absolute right-1 top-1">DAI</span>
+                <input
+                    v-model="userInput"
+                    class="Input"
+                    :class="{ Error: error }"
+                    :disabled="disabled"
+                    @focus="hideTotalPrice()"
+                    @blur="validateFinished()"
+                />
+            </label>
+        </Tooltip>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import { Tooltip } from 'ant-design-vue';
+import FormatCurrency from './FormatCurrency.vue';
+
+export default Vue.extend({
+    components: { FormatCurrency, Tooltip },
+    props: {
+        totalPrice: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        minimumDepositDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        value: {
+            type: Object as Vue.PropType<BigNumber> | undefined,
+            default: undefined,
+        },
+        disabled: {
+            type: Boolean,
+            default: false,
+        },
+    },
+    data() {
+        return {
+            userInput: undefined as string | undefined,
+            error: undefined as string | undefined,
+        };
+    },
+    computed: {
+        showErrorPopup(): boolean {
+            return !!this.error;
+        },
+    },
+    watch: {
+        value(newVal: BigNumber | undefined): void {
+            if (!newVal) {
+                this.resetToTotal();
+                return;
+            }
+            if (newVal.isEqualTo(this.totalPrice)) {
+                this.resetToTotal();
+            }
+            if (newVal.isEqualTo(this.minimumDepositDai)) {
+                this.userInput = newVal.toString();
+            }
+        },
+        error(newVal): void {
+            this.$emit('error', newVal);
+        },
+        userInput(newVal, oldVal) {
+            this.validateInput(newVal, oldVal);
+        },
+    },
+    methods: {
+        validateInput(newVal: string, oldVal: string): void {
+            this.error = undefined;
+            if (!newVal) {
+                return;
+            }
+            const value = new BigNumber(newVal);
+            if (isNaN(value.toNumber())) {
+                this.userInput = oldVal;
+            } else if (value.isGreaterThan(this.totalPrice)) {
+                this.error = `The bidding amount can not be greater than ${this.totalPrice.toFixed(2)} DAI`;
```

Comment:
I would not mix error logic with user input logic. Error logic can be refactored to a separate computed property